Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingFeatureProcessor.cpp

```diff
@@ -1067,19 +1126,65 @@ namespace AZ
-            bool enabled = true;
+            // only enable the RayTracingAccelerationStructurePass for each device on the first pipeline in this scene, this will avoid
+            // multiple updates to the same AS
             if (changeType == RPI::SceneNotification::RenderPipelineChangeType::Added
                 || changeType == RPI::SceneNotification::RenderPipelineChangeType::Removed)
             {
-                AZ::RPI::PassFilter passFilter = AZ::RPI::PassFilter::CreateWithPassName(AZ::Name("RayTracingAccelerationStructurePass"), GetParentScene());
-                AZ::RPI::PassSystemInterface::Get()->ForEachPass(passFilter, [&enabled](AZ::RPI::Pass* pass) -> AZ::RPI::PassFilterExecutionFlow
+                AZ::RPI::Pass* firstRayTracingAccelerationStructurePass{ nullptr };
+                auto rayTracingDeviceMask{ RHI::RHISystemInterface::Get()->GetRayTracingSupport() };
+                AZ::RHI::MultiDevice::DeviceMask devicesToAdd{ rayTracingDeviceMask };
+
+                AZ::RPI::PassFilter passFilter =
+                    AZ::RPI::PassFilter::CreateWithTemplateName(AZ::Name("RayTracingAccelerationStructurePassTemplate"), GetParentScene());
+                AZ::RPI::PassSystemInterface::Get()->ForEachPass(
+                    passFilter,
+                    [&devicesToAdd, &firstRayTracingAccelerationStructurePass, &rayTracingDeviceMask](
+                        AZ::RPI::Pass* pass) -> AZ::RPI::PassFilterExecutionFlow
                     {
-                        pass->SetEnabled(enabled);
-                        enabled = false;
+                        if (!firstRayTracingAccelerationStructurePass)
+                        {
+                            firstRayTracingAccelerationStructurePass = pass;
+                        }
+
+                        // we always set an invalid device index to the first available device
+                        if (pass->GetDeviceIndex() == RHI::MultiDevice::InvalidDeviceIndex)
+                        {
+                            pass->SetDeviceIndex(az_ctz_u32(AZStd::to_underlying(rayTracingDeviceMask)));
+                        }
+
+                        auto mask = RHI::MultiDevice::DeviceMask(AZ_BIT(pass->GetDeviceIndex()));
+
+                        // only have one RayTracingAccelerationStructurePass per device
+                        pass->SetEnabled((mask & devicesToAdd) != RHI::MultiDevice::NoDevices);
+                        devicesToAdd &= ~mask;
 
                         return AZ::RPI::PassFilterExecutionFlow::ContinueVisitingPasses;
                     });
+
+                // we only add the passes on the other devices if the pipeline contains one in the first place
+                if (firstRayTracingAccelerationStructurePass)
+                {
+                    // add passes for the remaining devices
```

Comment:
This preserves the current behavior where we also add scopes for all devices. How would you suggest making this data driven? The issue is that like with usual mesh data, we don't really know where we want to render the scene (with or without ray-tracing), so we want to keep the same behavior that model data is available on all devices. Alternatively, you could of course enforce people to add a pass for each device in their pipeline, but besides the additional work, that would be difficult, since you might want your pipeline to run on machines with different numbers of devices and not all might be ray-tracing capable.

Reply:
This is fine for now. I was thinking more in the sense that we could add a flag that needs to be enabled/disables in the render pipeline for this functionality.